File: examples/current.rb
#! /usr/bin/env ruby
require 'socket'
require 'json'

socket = TCPSocket.open('127.0.0.1', 9001)
socket.print "current\n"

song = socket.gets
socket.close()

if (song == 'null')
    exit
end

song = JSON.parse(song)

if (!song['id'].empty?)
    print "%s - " % song['id']
end

if (!song['title'].empty?)
    print "%s " % song['title']
end

if (!song['album'].empty?)
    print "(%s) " % song['album']
end

if (!song['artist'].empty?)
    print "{%s}" % song['artist']
end

puts ""

File: examples/current.pl
#! /usr/bin/env perl
use strict;
use warnings;
use IO::Socket;
use JSON;

my $socket = new IO::Socket::INET(
    PeerAddr => '127.0.0.1',
    PeerPort => 9001
);

print $socket 'current', "\n";
my $song = <$socket>;
close $socket;

my $json = new JSON()->allow_nonref(1);
$song    = $json->decode($song, { utf8 => 1 });

if ($song->{state} eq 'pause') {
    print "PAUSED. ";
}

if ($song->{id}) {
    print "$song->{id} - ";
}
if ($song->{title}) {
    print "$song->{title} ";
}

if ($song->{album}) {
    print "($song->{album}) "
}
if ($song->{artist}) {
    print "{$song->{artist}}";
}

print "\n";
